File: 1/ruby/part1.rb
def calibration_sum(document)
  document.lines.sum do |line|
    Integer(first_digit(line) + first_digit(line.reverse))
  end
end

def first_digit(line)
  line.each_char do |char|
    if char.ord >= '0'.ord && char.ord <= '9'.ord
      return char
    end
  end
end

document = File.read('../input')
puts calibration_sum(document)

File: 1/rust/src/part1.rs
use std::fs::File;
use std::io::{self, BufRead};
use std::iter::Iterator;

pub fn run() {
    let input_file = File::open("../input").unwrap();
    let lines = io::BufReader::new(input_file);
    let sum: u32 = lines.lines()
        .filter_map( |line| Some(line_value(line.ok()?)) )
        .map( |v| v.unwrap() )
        .sum();
    println!("{}", sum)
}

fn line_value(line: String) -> Option<u32> {
    let first_digit = line.chars().find(|c| c.is_ascii_digit())?.to_digit(10)?;
    let last_digit = line.chars().rev().find(|c| c.is_ascii_digit())?.to_digit(10)?;
    Some(10 * first_digit + last_digit)
}
